prow/tide/tide.go

```diff
@@ -756,6 +763,67 @@ func (c *Controller) takeAction(sp subpool, batchPending, successes, pendings, n
 	return Wait, nil, nil
 }
 
+// changedFilesAgent queries and caches the names of files changed by PRs.
+// Cache entries expire if they are not used during a sync loop or if
+// the PR branch is changed.
+type changedFilesAgent struct {
+	ghc         githubClient
+	changeCache map[string][]string
+	// nextChangeCache caches file change info that is relevant this sync for use next sync.
+	// This becomes the new changeCache when prune() is called at the end of each sync.
+	nextChangeCache map[string][]string
+	sync.Mutex
+}
+
+// prChanges gets the files changed by the PR, either from the cache or by
+// querying GitHub.
+func (c *changedFilesAgent) prChanges(pr *PullRequest) ([]string, error) {
+	cacheKey := fmt.Sprintf("%s/%s#%d:%s",
+		string(pr.Repository.Owner.Login),
+		string(pr.Repository.Name),
+		int(pr.Number),
+		string(pr.HeadRefOID),
+	)
+
+	changedFiles, ok := c.changeCache[cacheKey]
+	c.Lock()
```

Comment:
We don't lock for `c.changeCache` since we assume the only write is in `prune()` and architecturally we can't race with that? Could use a comment